File: ios/Demo/MekuriDemo/DemoInk.swift
import SwiftUI

/// Palette, strokes and type shared by every drawn page: ink on off-white
/// paper, one accent for what is live.
enum DemoInk {
    static let paper = Color(red: 0.97, green: 0.95, blue: 0.90)
    static let ink = Color(red: 0.12, green: 0.11, blue: 0.10)
    static let accent = Color(red: 0.80, green: 0.22, blue: 0.16)

    static let panelStroke: CGFloat = 3
    static let bubbleStroke: CGFloat = 2.5
    static let gutter: CGFloat = 12
    static let margin: CGFloat = 18

    static func sfx(_ size: CGFloat) -> Font {
        .system(size: size, weight: .black, design: .rounded)
    }

    static func narration(_ size: CGFloat) -> Font {
        .system(size: size, weight: .regular, design: .serif)
    }

    static func speech(_ size: CGFloat) -> Font {
        .system(size: size, weight: .semibold, design: .rounded)
    }

    static func title(_ size: CGFloat) -> Font {
        .system(size: size, weight: .black, design: .serif)
    }
}

File: android/demo/src/main/kotlin/studio/anicanon/mekuri/demo/DemoInk.kt
package studio.anicanon.mekuri.demo

import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/** Palette, strokes and type shared by every drawn page. */
object DemoInk {
    val Paper = Color(0.97f, 0.95f, 0.90f)
    val Ink = Color(0.12f, 0.11f, 0.10f)
    val Accent = Color(0.80f, 0.22f, 0.16f)
    val Backdrop = Color(0.12f, 0.12f, 0.12f)

    val PanelStroke = 3.dp
    val BubbleStroke = 2.5.dp
    val Gutter = 12.dp
    val Margin = 18.dp

    /** Page type scales against this reference page width in dp. */
    const val ReferencePage = 400f

    fun sfx(size: Float) = TextStyle(
        fontSize = size.sp,
        fontWeight = FontWeight.Black,
        fontFamily = FontFamily.Default,
        color = Ink,
    )

    fun narration(size: Float) = TextStyle(
        fontSize = size.sp,
        fontWeight = FontWeight.Normal,
        fontFamily = FontFamily.Serif,
        color = Ink,
    )

    fun speech(size: Float) = TextStyle(
        fontSize = size.sp,
        fontWeight = FontWeight.SemiBold,
        fontFamily = FontFamily.Default,
        color = Ink,
    )

    fun title(size: Float) = TextStyle(
        fontSize = size.sp,
        fontWeight = FontWeight.Black,
        fontFamily = FontFamily.Serif,
        color = Ink,
    )

    fun label(size: Float) = TextStyle(
        fontSize = size.sp,
        fontWeight = FontWeight.SemiBold,
        fontFamily = FontFamily.Default,
        color = Ink,
    )
}
